server/controllers/api/videos.js

```diff
@@ -622,3 +631,26 @@ function reportVideoAbuse (req, res, finalCallback) {
     return finalCallback(null)
   })
 }
+
+function addVideoToBlacklist (req, res, next) {
+  const videoInstance = res.locals.video
+
+  if (videoInstance.isOwned() === false) {
+    db.BlacklistedVideo.create({
+      remoteId: videoInstance.remoteId,
+      localId: videoInstance.id,
+      remotePodId: videoInstance.Author.Pod.id
+    })
+    .asCallback(function (err) {
+      if (err) {
+        logger.error('Errors when blacklisting video ', { error: err })
+        return next(err)
+      }
+
+      return res.type('json').status(204).end()
+    })
+    return
```

Comment:
I have left `return res.type('json').status(204).end()` because I get an error on the client when blacklisting a video otherwise. To be more precise, everything works fine, the video _is_ blacklisted but the notification shows an error status.